server/routes/friendLists/friendLists.route.js

```diff
@@ -0,0 +1,101 @@
+const express = require('express');
+const passport = require('passport');
+const mongoose = require('mongoose');
+const router = express.Router({ mergeParams: true });
+const FriendList = require('../../models/friendList');
+
+// All calls respond with the updated array of lists
+function getUserLists(userId) {
+  return new Promise((resolve, reject) => {
+    FriendList.find({ userId }, (err, docs) => {
+      if (err) reject(err);
+      FriendList.populate(docs, {
+        path: 'friendIds',
+        select: ['name', 'active'],
+      })
+        .then((result) => resolve(result))
+        .catch((err) => reject(err));
+    });
+  });
+}
+
+// @route:  GET /friendLists
+// @desc:   Get all user's friends lists
+// @access: Private
+router.get(
+  '/',
+  passport.authenticate('jwt', { session: false }),
+  async (req, res, next) => {
+    getUserLists(req.user._id)
+      .then((result) => {
+        res.json(result);
+      })
+      .catch(next);
```

Comment:
I'm not sure how this would behave in the frontend (as we discussed).

But it's worth noting that this method of catching errors is different than what is done in other routes.

I don't personally mind it, but it may be better to propose this change in a PR that looks at #79, which I see that you're assigned to.

Reply:
I'll do this:
Before merging this commit, I'll work on that issue by refactoring the error handling (BE and FE). Then after we merge that PR I'll merge this one. How does that sound?

Reply:
Sounds good